packages/lib-utils/src/k8s/hooks/useK8sWatchResource.test.ts

```diff
@@ -0,0 +1,179 @@
+import { renderHook } from '@testing-library/react-hooks/native';
+import { Map as ImmutableMap } from 'immutable';
+import { useSelector } from 'react-redux';
+import type { K8sModelCommon, K8sResourceCommon } from '../../types/k8s';
+import type { WatchData } from './k8s-watch-types';
+import type { NoModelError } from './k8s-watcher';
+import { getWatchData, getReduxData } from './k8s-watcher';
+import { useDeepCompareMemoize } from './useDeepCompareMemoize';
+import { useK8sModel } from './useK8sModel';
+import { useK8sWatchResource } from './useK8sWatchResource';
+import { useModelsLoaded } from './useModelsLoaded';
+import type { WatchK8sResource } from './watch-resource-types';
+
+const watchedResourceMock: WatchK8sResource = {
+  isList: false,
+  groupVersionKind: {
+    group: 'appstudio.redhat.com',
+    version: 'v1alpha1',
+    kind: 'Application',
+  },
+  name: 'test',
+  namespace: 'test-ns',
+};
+
+const resourceModelMock: K8sModelCommon | undefined = {
+  apiGroup: 'appstudio.redhat.com',
+  apiVersion: 'v1alpha1',
+  crd: true,
+  kind: 'Application',
+  namespaced: true,
+  plural: 'applications',
+};
+
+const resourceDataMock: K8sResourceCommon = {
+  apiVersion: 'appstudio.redhat.com/v1alpha1',
+  kind: 'Application',
+  metadata: {
+    annotations: { finalizeCount: '0', test: 'patched-value' },
+    creationTimestamp: '2022-04-29T13:41:21Z',
+    finalizers: ['application.appstudio.redhat.com/finalizer'],
+    generation: 1,
+    name: 'test',
+    namespace: 'vnambiar',
+    resourceVersion: '414309692',
+    uid: '602ad43f-1a71-4e71-9314-d93bffbc0762',
+  },
+  spec: { appModelRepository: { url: '' }, displayName: '', gitOpsRepository: { url: '' } },
+  status: {
+    conditions: [
+      {
+        lastTransitionTime: '2022-04-29T13:41:22Z',
+        message: 'Application has been successfully created',
+        reason: 'OK',
+        status: 'True',
+        type: 'Created',
+      },
+    ],
+    devfile:
+      'metadata:\\n  attributes:\\n    appModelRepository.context: /\\n    appModelRepository.url: https://github.com/redhat-appstudio-appdata/-vnambiar-touch-drive\\n    gitOpsRepository.context: /\\n    gitOpsRepository.url: https://github.com/redhat-appstudio-appdata/-vnambiar-touch-drive\\nschemaVersion: 2.1.0\\n',
+  },
+};
+
+let undefinedModelMock: K8sModelCommon | undefined;
+
+// Mock modules
+
+jest.mock('./k8s-watcher', () => ({
```

Comment:
@vidyanambiar I'll need to look at our Jest config, but we should ensure these mocks are cleared/reset between test files to prevent cross-contamination.